### PR TITLE
WIP: add ceph source CRD

### PR DESCRIPTION
**Pull request by yuvalif**
Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

Implementing ceph source CRD, controller etc. following [this guide](https://github.com/knative/docs/tree/master/docs/eventing/samples/writing-event-source)

- [ ] do end2end test
- [ ] write documentation
- [ ] write unit tests
- [ ] update dependencies after https://github.com/knative/eventing-contrib/issues/1571 is fixed
- [ ] update release notes

### CODE REVIEW

File: ceph/pkg/reconciler/ceph/controller.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+	"github.com/kelseyhightower/envconfig"
+	"k8s.io/client-go/tools/cache"
+
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+
+	"knative.dev/eventing-contrib/ceph/pkg/reconciler"
+
```
Comment (mattmoor) [inline ceph/pkg/reconciler/ceph/controller.go:34]: Format Go code:
```suggestion

	cephsourceinformer "knative.dev/eventing-contrib/ceph/pkg/client/injection/informers/sources/v1alpha1/cephsource"
	"knative.dev/eventing-contrib/ceph/pkg/client/injection/reconciler/sources/v1alpha1/cephsource"
```

File: ceph/pkg/reconciler/ceph/controller.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	eventingclient "knative.dev/eventing/pkg/client/injection/client"
+	sinkbindinginformer "knative.dev/eventing/pkg/client/injection/informers/eventing/v1beta1/broker"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	deploymentinformer "knative.dev/pkg/client/injection/kube/informers/apps/v1/deployment"
+	cephsourceinformer "knative.dev/eventing-contrib/ceph/pkg/client/injection/informers/sources/v1alpha1/cephsource"
+	"knative.dev/eventing-contrib/ceph/pkg/client/injection/reconciler/sources/v1alpha1/cephsource"
```
Comment (mattmoor) [inline ceph/pkg/reconciler/ceph/controller.go:40]: Format Go code:
```suggestion

```

File: ceph/pkg/reconciler/sinkbinding.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (mattmoor) [inline ceph/pkg/reconciler/sinkbinding.go:11]: Format Go code:
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"knative.dev/eventing-contrib/ceph/pkg/reconciler/resources"
```

File: ceph/pkg/reconciler/sinkbinding.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	"knative.dev/eventing/pkg/apis/sources/v1alpha2"
+	eventingclient "knative.dev/eventing/pkg/client/clientset/versioned"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+	"knative.dev/eventing-contrib/ceph/pkg/reconciler/resources"
```
Comment (mattmoor) [inline ceph/pkg/reconciler/sinkbinding.go:19]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -name '*.pb.go' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v .pb.go | grep -v wire_gen.go)`

**Comment by lberk** [comment]
@yuvalif we've moved the ceph source to its own sandbox repo: knative-sandbox/eventing-ceph

Could you please move this PR there? thanks!

**Comment by yuvalif** [comment]
> @yuvalif we've moved the ceph source to its own sandbox repo: [knative-sandbox/eventing-ceph](https://github.com/knative-sandbox/eventing-ceph)
> 
> Could you please move this PR there? thanks!

sure. will do

**Comment by yuvalif** [comment]
> @yuvalif we've moved the ceph source to its own sandbox repo: [knative-sandbox/eventing-ceph](https://github.com/knative-sandbox/eventing-ceph)
> 
> Could you please move this PR there? thanks!

@lberk we probably need to fix documentation [here](https://github.com/knative/docs/blob/master/docs/eventing/samples/writing-event-source/07-eventing-contrib.md)
as its refers to eventing-contrib repo and not the knative-sandbox org

**Comment by yuvalif** [comment]
this PR is closed in favor of: https://github.com/knative-sandbox/eventing-ceph/pull/5

**Pull request closed without merging**